packages/sdk/src/controllers/DataConnectorController.ts

```diff
@@ -0,0 +1,145 @@
+import { ConnectorConfigOptions, EditorAPI, EditorResponse, MetaData } from '../types/CommonTypes';
+import { getEditorResponseData } from '../utils/EditorResponseData';
+import { DataConnectorCapabilities, DataItem, DataPage, PageConfig } from '../types/DataConnectorTypes';
+
+/**
+ * The DataConnectorController is responsible for all communication regarding Data connectors.
+ * Methods inside this controller can be called by `window.SDK.DataConnector.{method-name}`
+ *
+ * The way CHILI Studio handles different sources of Data is called 'DataConnectors'. A DataConnectors is an
+ * implementation of a set of capabilities we need to interact with a certain Data Management system.
+ * In essence a connector is the combination of a Javascript snippet and some metadata. The Javascript snippet
+ * is loaded in the studio engine using a sandboxed Javascript execution engine (QuickJs). This allows us to
+ * execute the Data connector both on web using webassembly and on the server side during e.g. animation output
+ * generation.
+ * This controller is an interface to the running connector instance inside the studio engine.
+ */
+export class DataConnectorController {
+    /**
+     * @ignore
+     */
+    #editorAPI: EditorAPI;
+
+    /**
+     * @ignore
+     */
+    constructor(editorAPI: EditorAPI) {
+        this.#editorAPI = editorAPI;
+    }
+
+    /**
+     * Query a specific DataConnector for data using both specific PageConfig and the dynamic
+     * context as parameters.
+     * @param connectorId unique id of the Data connector
+     * @param config page configuration
+     * @param context context that will be available in the connector script.
+     * @returns a DataPage with an array of data objects
+     */
+    getPage = async (
+        connectorId: string,
+        config: PageConfig,
+        context: MetaData = {},
+    ): Promise<EditorResponse<DataPage<DataItem>>> => {
+        const res = await this.#editorAPI;
+        return res
+            .dataConnectorGetPage(connectorId, JSON.stringify(config), JSON.stringify(context))
+            .then((result) => getEditorResponseData<DataPage<InternalDataItem>>(result))
+            .then((resp) => {
+                const update: EditorResponse<DataPage<DataItem>> = { ...resp, parsedData: null };
+                if (resp.parsedData) {
+                    update.parsedData = {
+                        data: resp.parsedData.data.map((e: InternalDataItem) => this.mapInternalToDataItem(e)),
+                        continuationToken: resp.parsedData.continuationToken,
+                    };
+                }
+                return update;
+            });
+    };
+
+    /**
+     * Query a specific DataConnector for a data model using the dynamic context as parameter.
+     * @param connectorId unique id of the media connector
```

Comment:
There are more like this, please review the PR for other places where the wrong one is used.

```suggestion
     * @param connectorId unique id of the data connector
```